### PR TITLE
MDS-325 | Added new versioning/overview md files and changed raml to suit

### PR DESCRIPTION
**Pull request by matthewharris1**

### CODE REVIEW

File: app/uk/gov/hmrc/individualsemploymentsapi/controller/DocumentationController.scala [hunk 1]
```diff
@@ -32,7 +32,7 @@ class DocumentationController @Inject()(
   config: Configuration)
     extends BackendController(cc) {
   private lazy val privilegedWhitelistedApplicationIds =
-    config.getOptional[Seq[String]]("api.access.version-P1.0.whitelistedApplicationIds").getOrElse(Seq.empty)
+    config.getOptional[Seq[String]]("api.access.version-P2.0.whitelistedApplicationIds").getOrElse(Seq.empty)
```
Comment (ajchilts) [inline app/uk/gov/hmrc/individualsemploymentsapi/controller/DocumentationController.scala:35]: If we change this how does P1 documentation get served? HO will remain on V1 and so we need to keep the V1 thread. We need to add a new thread of docs for V2.

File: resources/public/api/conf/P1.0/application.raml [hunk 2]
```diff
@@ -1,16 +1,16 @@
-#%RAML 1.0
+#%RAML 2.0
 ---
 
 title: Individuals Employments
-version: P1.0
+version: P2.0
```
Comment (ajchilts) [inline resources/public/api/conf/P1.0/application.raml:5]: I don't think we should be changing V1 docs to V2?

File: resources/public/api/conf/P1.0/docs/overview.md [hunk 3]
```diff
@@ -1,13 +1,20 @@
-<p>This API provides individual's employment information from HM Revenue and Customs (HMRC).</p>
+# Overview
```
Comment (ajchilts) [inline resources/public/api/conf/P1.0/docs/overview.md:1]: Same here. This is V1 documentation and needs to stay as is. We need a V2.

**Review by ajchilts — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/individualsemploymentsapi/controller/DocumentationController.scala (hunk 1 above)
Comment (owen-opencast) [reply app/uk/gov/hmrc/individualsemploymentsapi/controller/DocumentationController.scala:35]: The documentation itself is served up from the documentation method, which dynamically generates the path to the documentation files. Not sure what this line is used for (it's pointed to by the /api/definition route)...

**Review by ajchilts — APPROVED** [review]

**Pull request merged**